### PR TITLE
nextcloud - limit access to php-fpm

### PR DESCRIPTION
**Pull request by szaimen**
Close https://github.com/nextcloud/all-in-one/issues/3172

### CODE REVIEW

File: Containers/nextcloud/entrypoint.sh [hunk 1]
```diff
@@ -37,6 +37,11 @@ if [ -n "$PHP_MAX_CHILDREN" ]; then
     sed -i "s/^pm.max_children =.*/pm.max_children = $PHP_MAX_CHILDREN/" /usr/local/etc/php-fpm.d/www.conf
 fi
 
+# Limit access to php-fpm
+# TODO
+grep -q ';listen.allowed_clients' /usr/local/etc/php-fpm.d/www.conf
+sed -i 's|;listen.allowed_clients.*|listen.allowed_clients = 127.0.0.1,::1#TODO|' /usr/local/etc/php-fpm.d/www.conf
```
Comment (szaimen) [inline Containers/nextcloud/entrypoint.sh:43]: todo, see https://github.com/nextcloud/all-in-one/issues/3172#issue-1853054017

Comment (szaimen) [reply Containers/nextcloud/entrypoint.sh:43]: And if we introduce dig we might also adjust the collabora logic to use dig insteatd of php

**Comment by szaimen** [comment]
@Zoey2936 does this look good to you? :)

File: Containers/nextcloud/start.sh [hunk 2]
```diff
@@ -25,6 +25,23 @@ if [ -f "/var/www/html/config/config.php" ]; then
...
+IPv6_ADDRESS_MASTERCONTAINER="$(dig nextcloud-aio-mastercontainer AAAA +short | grep '^[0-9a-f:]\+$' | sort | head -n1)"
+IPv4_ADDRESS_NOTIFY_PUSH="$(dig nextcloud-aio-notify-push A +short | grep '^[0-9.]\+$' | sort | head -n1)"
+IPv6_ADDRESS_NOTIFY_PUSH="$(dig nextcloud-aio-notify-push AAAA +short | grep '^[0-9a-f:]\+$' | sort | head -n1)"
+IPv4_ADDRESS_DSP="$(dig nextcloud-aio-docker-socket-proxy A +short | grep '^[0-9.]\+$' | sort | head -n1)"
+IPv6_ADDRESS_DSP="$(dig nextcloud-aio-docker-socket-proxy AAAA +short | grep '^[0-9a-f:]\+$' | sort | head -n1)"
+set +x
+
+sed -i "s|^;listen.allowed_clients|listen.allowed_clients|" /usr/local/etc/php-fpm.d/www.conf
+sed -i "s|listen.allowed_clients.*|listen.allowed_clients = 127.0.0.1,::1,$IPv4_ADDRESS_APACHE,$IPv6_ADDRESS_APACHE,$IPv4_ADDRESS_MASTERCONTAINER,$IPv6_ADDRESS_MASTERCONTAINER,$IPv4_ADDRESS_NOTIFY_PUSH,$IPv6_ADDRESS_NOTIFY_PUSH,$IPv4_ADDRESS_DSP,$IPv6_ADDRESS_DSP|" /usr/local/etc/php-fpm.d/www.conf
+sed -i "listen.allowed_clients/s/,,//" /usr/local/etc/php-fpm.d/www.conf
```
Comment (Zoey2936) [inline Containers/nextcloud/start.sh:41]:
```suggestion
sed -i "listen.allowed_clients/s/,,/,/" /usr/local/etc/php-fpm.d/www.conf
```

Comment (szaimen) [reply Containers/nextcloud/start.sh:41]: right, thanks!

**Pull request merged**